### DISCUSSION TITLE
Can i create a single ScenegraphLayer to render different gltf objects?

### DISCUSSION THREAD

**Discussion by linze99** [visgl/deck.gl#7067] [Q&A] [answered]
Can i create a single ScenegraphLayer to render different gltf objects?
For example,i have an array of objects likes
```js
[
  {
    geometry: {
      coordinates: [lng,lat],
      type: "Point",
    },
    type: "Feature",
    properties: {
      name: "name",
      model: "/download/model1.glb",// there are some different model files,each data specify a model file here.
      scale: 1,
      orientation: [0,90,90],
    },
  },
,...]
```
then we can create a ScenegraphLayer to render every data with the model property  like these properties:
```js
getScale: (d)=>d.properties.scale,
getOrientation:(d)=>d.properties.orientation,
getScenegraph: (d)=>d.properties.model,
```

**Comment by ibgreen** [comment, marked as answer]
No, this is not currently supported.

In a way this would go against the design of deck.gl where the idea is that a single GPU draw call draws many similar objects for each layer.

It would not be impossible to create a layer that issues multiple draw calls but it would have some performance penalty as it would need to do things like splitting data into multiple arrays, one per model. It would be quite equivalent to the app itself using multiple ScenegraphLayers.
